presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -573,6 +574,10 @@ void PrestoServer::run() {
   addAdditionalPeriodicTasks();
   periodicTaskManager_->start();
 
+  if (folly::Singleton<PeriodicMemoryChecker>::try_get()) {
```

Comment:
Can you add two virtual methods in PrestoServer
```
addMemoryCheckerPeriodicTask -> call on server startup
stopMemoryCheckerPeriodicTask -> call on server shutdown
And add a unique_ptr of base PeriodicMemoryChecker in PrestoServer?
```
That's how meta works internally. I can make a change in Meta to overload the two methods. Thanks!

Reply:
cc @majetideepak

Reply:
@xiaoxmeng I have added the two virtual methods in PrestoServer, please take a look when you can, thanks!

Reply:
@xiaoxmeng The unique_ptr<PeriodicMemoryChecker> approach would pull in additional headers that we can avoid with the folly::singleton approach. Do you have any concerns?